python/tank/bootstrap/cached_configuration.py

```diff
@@ -399,13 +418,14 @@ def _log_core_information(self):
             )
 
     @property
-    def has_local_bundle_cache(self):
+    def requires_dynamic_bundle_caching(self):
         """
-        If True, indicates that pipeline configuration has a local bundle cache. If False, it
-        depends on the global bundle cache.
+        If True, indicates that pipeline configuration relies on dynamic caching
+        of bundles to operate. If False, the configuration has its own bundle
+        cache.
         """
         # CachedConfiguration always depend on the global bundle cache.
-        return False
+        return True
```

Comment:
If everything is packaged, why is the dynamic bundle caching necessary? Arguably, if a package is missing it should probably be a bug, right?

The descriptor's get_bundle_cache_override_folder() method could actually be used here. If that method returns something, it mean no caching is required. `requires_dynamic_bundle_caching()` can actually move to the base class at that point since the descriptor is the entity responsible for knowing if the global bundle cache needs to be populated or not.

Reply:
This is the cached config, so we don't assume that all bundles are packaged with the config. Maybe I'm misunderstanding your first question. 

What about cases where the bundle cache override folder exists, but isn't fully populated? Would it be reasonable for someone to build a partially populated bundle cache. Imagine you want to package bundles that aren't widely available for download (like studio-specific apps/engines) while allowing the usual toolkit bundles to be downloaded during bootstrap. Seems reasonable, no?

Reply:
Good points.